legend-engine-xt-changetoken-compiler/src/main/java/org/finos/legend/engine/changetoken/generation/GenerateCastUtil.java

```diff
@@ -152,4 +152,43 @@ private static int toInt(Object key)
     {
         return (key instanceof Integer) ? (Integer) key : Integer.valueOf((String)key);
     }
+
+    private static String pathToString(List<Object> path)
+    {
+        return path.stream().map(Object::toString).collect(Collectors.joining("/"));
+    }
+
+    public static String convertField_IntegerToString(List<Object> path, Object value)
+    {
+        try
+        {
+            if (!(value instanceof Integer))
+            {
+                throw new RuntimeException("Expected an Integer, not " + value);
+            }
+
+            return Integer.toString((Integer) value);
+        }
+        catch (RuntimeException ex)
+        {
+            throw new RuntimeException(ex + " at path " + pathToString(path), ex);
+        }
+    }
+
+    public static Integer convertField_StringToInteger(List<Object> path, Object value)
```

Comment:
If you're going to use these for indexes, then you should return int instead of Integer.
```suggestion
    public static int convertField_StringToInteger(List<Object> path, Object value)
```

Reply:
No, that's for type conversion. There's a different function for parseInt already